### PR TITLE
tests(intergration/proxy/balancer/healthchecks) fix some flaky tests

### PR DESCRIPTION
**Pull request by ADD-SP**
_FT-3267_

### CODE REVIEW

File: spec/helpers.lua [hunk 1]
```diff
@@ -1554,6 +1561,129 @@ local function wait_for_invalidation(key, timeout)
...
+--- Wait for all targets, upstreams, services, and routes update
+-- 
+-- NOTE: this function is not available for DBless-mode
+-- @function wait_for_all_config_update
+-- @tparam[opt=30] number timeout maximum time to wait
+-- @tparam[opt] number admin_client_timeout, to override the default timeout setting
+-- @tparam[opt] number forced_admin_port to override the default port of admin API
+-- @usage helpers.wait_timer()
+local function wait_for_all_config_update(timeout, admin_client_timeout, forced_admin_port)
+  timeout =  timeout and timeout or 30
```
Comment (chronolaw) [inline spec/helpers.lua:1573]: `timeout = timeout or 30`

Comment (ADD-SP) [reply spec/helpers.lua]: done

File: spec/helpers.lua [hunk 2]
```diff
@@ -1554,6 +1561,129 @@ local function wait_for_invalidation(key, timeout)
...
+  service_id = json.id
+
+  -----------------------------
+
+  res = client:post(string.format("/services/%s/routes", service_id), {
+    headers = {["Content-Type"] = "application/json"},
+    body = {
+      paths = { route_path },
+      strip_path = true,
+      path_handling = "v1",
```
Comment (chronolaw) [inline spec/helpers.lua:1637]: I think `path_handling` should be `v0`, because `v1` will be deprecated in the future.

File: spec/helpers.lua [hunk 3]
```diff
@@ -1554,6 +1561,129 @@ local function wait_for_invalidation(key, timeout)
...
+  local body = luassert.res_status(201, res)
+  local json = cjson.decode(body)
+
+  if not json.id then
+    error("failed to add dummy upstream")
+  end
+
+  upstream_id = json.id
+
+  -----------------------------
```
Comment (chronolaw) [inline spec/helpers.lua:1598]: This line could be more meaningful, such as `-- create a new target`.

File: spec/helpers.lua (hunk 2 above)
Comment (ADD-SP) [reply spec/helpers.lua]: done

File: spec/helpers.lua (hunk 3 above)
Comment (ADD-SP) [reply spec/helpers.lua]: done

**Review by hanshuebner — APPROVED** [review]

**Pull request merged**